controller/api/public/grpc_server.go

```diff
@@ -3,200 +3,55 @@ package public
 import (
 	"context"
 	"errors"
-	"fmt"
 	"runtime"
-	"time"
 
-	"github.com/golang/protobuf/ptypes/duration"
 	destinationPb "github.com/linkerd/linkerd2-proxy-api/go/destination"
-	"github.com/linkerd/linkerd2/controller/api/util"
-	publicPb "github.com/linkerd/linkerd2/controller/gen/public"
+	pb "github.com/linkerd/linkerd2/controller/gen/public"
 	"github.com/linkerd/linkerd2/controller/k8s"
-	pkgK8s "github.com/linkerd/linkerd2/pkg/k8s"
 	"github.com/linkerd/linkerd2/pkg/prometheus"
 	"github.com/linkerd/linkerd2/pkg/version"
-	pb "github.com/linkerd/linkerd2/viz/metrics-api/gen/viz"
-	healthcheckPb "github.com/linkerd/linkerd2/viz/metrics-api/gen/viz/healthcheck"
-	promv1 "github.com/prometheus/client_golang/api/prometheus/v1"
 	log "github.com/sirupsen/logrus"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/labels"
 )
 
-// PublicAPIServer specifies the interface the Public API server should implement
-// TODO: remove stuttering name when viz api code moves to /viz
-// nolint
-type PublicAPIServer interface {
-	publicPb.ApiServer
-	destinationPb.DestinationServer
-}
-
-// VizAPIServer specifies the interface the Viz metric API server should implement
-type VizAPIServer interface {
+// Server specifies the interface the Public API server should implement
+type Server interface {
 	pb.ApiServer
+	destinationPb.DestinationServer
 }
 
 type grpcServer struct {
-	prometheusAPI       promv1.API
 	destinationClient   destinationPb.DestinationClient
 	k8sAPI              *k8s.API
 	controllerNamespace string
 	clusterDomain       string
-	ignoredNamespaces   []string
 }
 
-type podReport struct {
-	lastReport              time.Time
-	processStartTimeSeconds time.Time
-}
-
-const (
-	podQuery                  = "max(process_start_time_seconds{%s}) by (pod, namespace)"
-	k8sClientSubsystemName    = "kubernetes"
-	k8sClientCheckDescription = "control plane can talk to Kubernetes"
-	// promClientSubsystemName    = "prometheus"
-	// promClientCheckDescription = "control plane can talk to Prometheus"
-)
-
 func newGrpcServer(
-	promAPI promv1.API,
 	destinationClient destinationPb.DestinationClient,
 	k8sAPI *k8s.API,
 	controllerNamespace string,
 	clusterDomain string,
-	ignoredNamespaces []string,
 ) *grpcServer {
 
 	grpcServer := &grpcServer{
-		prometheusAPI:       promAPI,
 		destinationClient:   destinationClient,
 		k8sAPI:              k8sAPI,
 		controllerNamespace: controllerNamespace,
 		clusterDomain:       clusterDomain,
-		ignoredNamespaces:   ignoredNamespaces,
 	}
 
 	pb.RegisterApiServer(prometheus.NewGrpcServer(), grpcServer)
```

Comment:
If you mean the `prometheusAPI` field, that still is used in the viz gRPC file (`viz/metrics-api/grpc_server.go`). The public-api should have no dependencies on prometheus.

Reply:
I think this is for instrumenting the server and exposing controller metrics (which can be scraped by prometheus)

Reply:
@alpeb Nope I'm referring to the `pb.RegisterApiServer(prometheus.NewGrpcServer(), grpcServer)` line 43.

And if that is already taken care of by the admin server then maybe this can be removed then?

Reply:
Ok then that does look like something about the prom metrics that are being served. But if I comment that out, the metrics served look the same (go and http related metrics). I honestly have no idea what that does, and it's not documented in those functions either :man_shrugging: